lib/ReactViews/Story/StoryPanel.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import ObserveModelMixin from '../ObserveModelMixin';
+import parseCustomHtmlToReact from '../Custom/parseCustomHtmlToReact';
+import { Small, Medium } from '../Generic/Responsive';
+import Icon from "../Icon.jsx";
+import { Swipeable } from 'react-swipeable';
+import Styles from './story-panel.scss';
+
+// given a nowviewing item, build its path in the data catalog
+export function buildPath(item) {
+      if(!item.parent) {
```

Comment:
Use the catalog item's `uniqueId` property, which is basically the same thing but more resistant to catalog items being renamed and moved.